edweiss-firebase/functions/src/functions/lectures/markQuestionAsAnswered.ts

```diff
@@ -0,0 +1,45 @@
+/**
+ * @file markQuestionAsAnswered.ts
+ * @description Cloud function for marking a question as answered
+ * @author Adamm Alaoui
+ */
+
+// ------------------------------------------------------------
+// --------------- Import Modules & Components ----------------
+// ------------------------------------------------------------
+
+import LectureDisplay from 'model/lectures/lectureDoc';
+import { onAuthentifiedCall } from 'utils/firebase';
+import { clean, CollectionOf, getDocumentAndRef } from 'utils/firestore';
+import { fail, ok } from 'utils/status';
+
+// ------------------------------------------------------------
+// ---------   Mark Question Answered Cloud Function   --------
+// ------------------------------------------------------------
+
+export const markQuestionAsAnswered = onAuthentifiedCall(LectureDisplay.Functions.markQuestionAsAnswered, async (userId, args) => {
+    if (!args.id || args.id.length == 0)
+        return fail("invalid_id");
+
+    const [question, ref] = await getDocumentAndRef(CollectionOf<LectureDisplay.Question>(`courses/${args.courseId}/lectures/${args.lectureId}/questions`), args.id);
+
+    if (question == undefined)
+        return fail("error_firebase");
+
+    const updatedQuestion: Partial<LectureDisplay.Question> = {
+        anonym: question.anonym,
+        likes: question.likes,
+        postedTime: question.postedTime,
+        text: question.text,
+        userID: question.userID,
+        username: question.username,
+        answered: args.answered,
+    };
```

Comment:
You didn't check if the user has acces to the answer,  it think, it's a security issue

Reply:
So I do not understand your point. This cloud function is only meant to be used by the professor, the username is a string to display it on the `QuestionDisplay`. It has nothing to do with security. The question model state that the `answered` boolean is for the teacher to know what question remains to be answered. Is it clearer this way ?

Reply:
Ok i understand, i was not sure, so i prefer to ask

Reply:
Great, I am here for this 😉